src/parsers.rs

```diff
@@ -150,3 +472,116 @@ pub(crate) fn parse_time(source: &str) -> TemporalResult<TimeRecord> {
 }
 
 // TODO: ParseTimeZoneString, ParseZonedDateTimeString
+
+#[cfg(test)]
+mod tests {
+    use crate::parsers::{FormattableTime, Precision};
+
+    use super::{write_u32_to_ascii_digits, FormattableDate, FormattableOffset};
+
+    #[test]
+    fn offset_string() {
+        let offset = FormattableOffset {
+            sign: crate::Sign::Positive,
+            time: FormattableTime {
+                hour: 4,
+                minute: 0,
+                second: 0,
+                nanosecond: 0,
+                precision: Precision::Minute,
+                include_sep: true,
+            },
+        };
+        assert_eq!(offset.to_string(), "+04:00");
```

Comment:
Nit: Use `assert_writeable_eq` instead of `assert_eq` in order to test the Writeable impl more thoroughly (such as the correctness of `writeable_length_hint`)